unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/ExploreV2/Scripts/MainMenu/ExploreV2Menu/ExploreV2MenuComponentController.cs

```diff
@@ -1,95 +1,87 @@
+    private UserProfile ownUserProfile => UserProfile.GetOwnUserProfile();
+    private BaseVariable<bool> isPlacesAndEventsSectionInitialized => DataStore.i.exploreV2.isPlacesAndEventsSectionInitialized;
+    private BaseVariable<bool> isPromoteChannelsToastVisible => DataStore.i.channels.isPromoteToastVisible;
+
+    private RectTransform topMenuTooltipReference => view.currentTopMenuTooltipReference;
+    private RectTransform placesAndEventsTooltipReference => view.currentPlacesAndEventsTooltipReference;
+    private RectTransform backpackTooltipReference => view.currentBackpackTooltipReference;
+    private RectTransform mapTooltipReference => view.currentMapTooltipReference;
+    private RectTransform questTooltipReference => view.currentQuestTooltipReference;
+    private RectTransform settingsTooltipReference => view.currentSettingsTooltipReference;
+    private RectTransform profileCardTooltipReference => view.currentProfileCardTooltipReference;
 
     public void Initialize()
     {
         sectionsVariables = new Dictionary<ExploreSection, (BaseVariable<bool>, BaseVariable<bool>)>
         {
-            { ExploreSection.Explore, (isPlacesAndEventsSectionInitialized,  placesAndEventsVisible) },
-            { ExploreSection.Backpack, (isAvatarEditorInitialized,  avatarEditorVisible) },
-            { ExploreSection.Map, (isNavmapInitialized,  navmapVisible) },
-            { ExploreSection.Quest, (isQuestInitialized,  questVisible) },
-            { ExploreSection.Settings, (isSettingsPanelInitialized,  settingsVisible) },
+            { ExploreSection.Explore, (isPlacesAndEventsSectionInitialized, placesAndEventsVisible) },
+            { ExploreSection.Backpack, (isAvatarEditorInitialized, avatarEditorVisible) },
+            { ExploreSection.Map, (isNavmapInitialized, navmapVisible) },
+            { ExploreSection.Quest, (isQuestInitialized, questVisible) },
+            { ExploreSection.Settings, (isSettingsPanelInitialized, settingsVisible) },
         };
+
         sectionsByInitVar = sectionsVariables.ToDictionary(pair => pair.Value.initVar, pair => pair.Key);
-        sectionsByVisiblityVar = sectionsVariables.ToDictionary(pair => pair.Value.visibilityVar, pair => pair.Key);
+        sectionsByVisibilityVar = sectionsVariables.ToDictionary(pair => pair.Value.visibilityVar, pair => pair.Key);
 
         mouseCatcher = SceneReferences.i?.mouseCatcher;
         exploreV2Analytics = CreateAnalyticsController();
+
         view = CreateView();
         SetVisibility(false);
 
-        DataStore.i.realm.realmName.OnChange += UpdateRealmInfo;
-        UpdateRealmInfo( DataStore.i.realm.realmName.Get());
-
-        DataStore.i.realm.realmsInfo.OnSet += UpdateAvailableRealmsInfo;
-        UpdateAvailableRealmsInfo(DataStore.i.realm.realmsInfo.Get());
+        realmController = new ExploreV2ComponentRealmsController(DataStore.i.realm, view);
+        realmController.Initialize();
```

Comment:
It was just small improvement of the code and of course not finished one. From my point of view, it would make much more sense to create separate `ExploreV2ComponentRealmsView` for better testability. However, how they should be composed is a different question. As for now, it is quite clear composition in the hierarchical way - `exploreV2ComponentController `is composed from `exploreV2ComponentRealmsController` and other components and fully controls it's life cycle. Creation is clear and readable and all realms logic is grouped. 

But I don't see benefits of presenting additional abstraction layer (an interface `IExploreV2ComponentRealmsController`) and also having it injected somehow on the top. It will only brings additional complexity without that much benefits. Being a big fun of KISS and YAGNI principle, I will leave it as is for now and see in the future refactoring if we will need to extract this interface and replace Composition relation of `ExploreV2ComponentController ` to `exploreV2ComponentRealmsController` by an Aggregation.